hal/armv7m/imxrt/117x/imxrt.c

```diff
@@ -304,14 +366,17 @@ void _imxrt_init(void)
 	imxrt_common.wdog2 = (void *)0x40034000;
 	imxrt_common.wdog3 = (void *)0x40038000;
 	imxrt_common.src = (void *)0x40c04000;
-	imxrt_common.iomux_snvs = (void *)0x40c94000;
-	imxrt_common.iomux_lpsr = (void *)0x40c08000;
+	imxrt_common.iomuxc_snvs = (void *)0x40c94000;
+	imxrt_common.iomuxc_lpsr = (void *)0x40c08000;
+	imxrt_common.iomuxc_lpsr_gpr = (void *)0x40c0c000;
 	imxrt_common.iomuxc_gpr = (void *)0x400e4000;
 	imxrt_common.iomuxc = (void *)0x400e8000;
 	imxrt_common.ccm = (void *)0x40cc0000;
 
 	imxrt_common.cpuclk = 640000000;
 
+	imxrt_common.cm4state = (((*(imxrt_common.src + src_scr) & 1u) != 0u) ? 1u : 0u);
```

Comment:
I think it's enough.
```suggestion
	imxrt_common.cm4state = *(imxrt_common.src + src_scr) & 1u;
```

Reply:
Agree it's "over-engineered" :D, I'll change it to your suggest.